gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -152,8 +153,27 @@ public static void movePath(FileSystem srcFs, Path src, FileSystem dstFs, Path d
     if (srcFs.getUri().equals(dstFs.getUri())) {
       renamePath(srcFs, src, dst);
     } else {
-      if (!FileUtil.copy(srcFs, src, dstFs, dst, true, false, dstFs.getConf())) {
-        throw new IOException(String.format("Failed to move %s to %s", src, dst));
+      boolean isSourceFileSystemLocal = srcFs instanceof LocalFileSystem;
```

Comment:
This won't cover `RawLocalFileSystem`. What's the benefit of differentiating here?